pkg/core/bundle.go

```diff
@@ -293,24 +220,24 @@ func implPublishMetadata(ctx context.Context, bundle *Bundle,
 }
 
 // Upload an bundle to archive
-func Upload(ctx context.Context, bundle *Bundle) error {
-	return implUpload(ctx, bundle, defaultBundleEntriesPerFile, nil)
+func Upload(ctx context.Context, bundle *Bundle, opts ...ListOption) error {
```

Comment:
with `ListOption` now documented as options pertaining to list operations (`bundle list`, `repo list`, etc.), using `ListOption`s to `Upload` is not something we'll want to ship as part of a stable api..

..i dunno if it's worth worrying about as part of this merge request or not.  moreover, i don't know if the change should be as shallow as using a different name from`ListOption` to describe the same `func(*Settings)` type or whether splitting hairs at the level of structures containing different types of options appears worthwhile.

the only strong opinion here is to have the documentation agree with actual usage by the time (if ever) we expect developers to use the `core` api independently of the datamon executable.

Reply:
This was added to avoid a global package level export symbol to enable memory profiling, nothing more

Reply:
I think `ListOption` will be renamed to simply `Option` in the near future

Reply:
> I think `ListOption` will be renamed to simply `Option` in the near future

i'd be alright with this.